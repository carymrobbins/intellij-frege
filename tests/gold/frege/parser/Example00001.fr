module Main where

import Data.List

main _ = do
  println foo
  println msg
  where
  foo = "bar"

msg = Data.List.head ["Hello World"]
