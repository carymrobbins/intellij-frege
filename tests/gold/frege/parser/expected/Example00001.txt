Frege file
  ModDcl
    ImpDcl
      PsiElement(CONID)('Main')
      PsiElement(WHERE)('where')
      PsiElement(IMPORT)('import')
      PsiElement(QUALIFIER)('Data')
      PsiElement(CONID)('List')
    FunDcl
      Vbl
        Simple
          PsiElement(VARID)('msg')
      Pats
        <empty list>
      PsiElement(CHAR)('=')
      App
        Vbl
          With2
            PsiElement(QUALIFIER)('Data')
            PsiElement(QUALIFIER)('List')
            PsiElement(VARID)('head')
        App
          App
            Con
              With1
                PsiElement(CHAR)('[')
            Lit
              PsiElement(STRCONST)('"Hello World"')
          Con
            With1
              PsiElement(CHAR)(']')
    FunDcl
      Vbl
        Simple
          PsiElement(VARID)('main')
      Pats
        Pat
          Vbl
            Simple
              PsiElement(VARID)('_')
      PsiElement(CHAR)('=')
      App
        Vbl
          Simple
            PsiElement(VARID)('println')
        Vbl
          Simple
            PsiElement(VARID)('msg')